include/mockturtle/views/cnf_view.hpp

```diff
@@ -71,18 +159,22 @@ struct cnf_view_params
  * added.  Therefore, a network cannot modify or delete nodes when wrapped in a
  * `cnf_view`.
  */
-template<typename Ntk, bill::solvers Solver = bill::solvers::glucose_41>
-class cnf_view : public Ntk
+template<typename Ntk, bool AllowModify, bill::solvers Solver>
```

Comment:
We need some tests in `cnf_view.cpp` for the case that `AllowModify = true` starting from an existing network, modified network, deleted node, ...